TrafficSimulator/Assets/RoadGenerator/Core/Runtime/Objects/RoadSystemGraph.cs

```diff
@@ -0,0 +1,209 @@
+                graphNode.Edges.Add(new GraphEdge(PreviouslyAddedNode, _currentCost));
+                Graph.Add(curr.Position.ToString(), graphNode);
+                PreviouslyAddedNode = graphNode;
+                _currentCost = 0f;
+                curr = curr.Next;
+            }
+
+        }
+        private float CalculateCost(float distance, float speedLimit = 1f)
+        {
+            return distance / speedLimit;
+        }
+    }
+
+    public static class RoadSystemNavigationGraph
+    {
+        private const string GRAPH_NODE_SPHERE_NAME = "Graph Node";
+        private const float GRAPH_NODE_SPHERE_SIZE = 15f;
+        private const float EDGE_LINE_WIDTH = 4f;
+        private const float GRAPH_LIFT = 20f;
+
+        /// <summary> Generates a graph representation for a road system </summary>
+        public static Dictionary<string, GraphNode> GenerateRoadSystemNavigationGraph(RoadSystem roadSystem)
+        {
+            // The road system graph, key is the positions string representation
+            Dictionary<string, GraphNode> roadSystemGraph = new Dictionary<string, GraphNode>();
+            // Loop through all roads in the road system
+            foreach(Road road in roadSystem.Roads)
+            {   
+                // Map the road into the graph
+                UpdateGraphForRoad(road, roadSystemGraph);
+
+            }
+            return roadSystemGraph;
+        }
+
+        /// <summary> Updates a graph for the given road   </summary>
+        private static void UpdateGraphForRoad(Road road,  Dictionary<string, GraphNode> roadSystemGraph)
+        {
+            GraphNode[] roadNavigationGraph = new GraphNode[road.NavigationGraph.Graph.Values.Count];
+            road.NavigationGraph.Graph.Values.CopyTo(roadNavigationGraph, 0);
+            for (var i = 0; i < roadNavigationGraph.Length; i++)
+            {
+                if (roadSystemGraph.ContainsKey(roadNavigationGraph[i].RoadNode.Position.ToString()))
+                {
+                    GraphNode node = roadSystemGraph[roadNavigationGraph[i].RoadNode.Position.ToString()];
+                    node.Edges.AddRange(roadNavigationGraph[i].Edges);
+                    roadNavigationGraph[i] = node;
+
+                    continue;
+                }
+                roadSystemGraph.Add(roadNavigationGraph[i].RoadNode.Position.ToString(), roadNavigationGraph[i]);
+            }
+        }
+        
+         /// <summary> Draws the graph </summary>
+        public static void DrawGraph(RoadSystem roadSystem, Dictionary<string, GraphNode> roadGraph)
+        {
+            foreach (GraphNode node in roadGraph.Values)
+            {
+                // Create a new graph node sphere
+                GameObject nodeObject = GameObject.CreatePrimitive(PrimitiveType.Sphere);
+                
+                // Name it and place it on the correct location
+                nodeObject.name = GRAPH_NODE_SPHERE_NAME;
```

Comment:
To avoid having to do this all in code, a GraphNode prefab should be added that is a sphere and has the correct material and settings so you only have to change the parent and position